### PR TITLE
chore(release): add travis next release script

### PR DESCRIPTION
**Pull request by christianalfoni**
- Updated repo-cooker to handle branch specific logic
- Updated travis file to run publish on master and next branch
- Dry run on both master and next, looks good!

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -29,11 +29,11 @@ after_success:
     npm run coverage:upload;
   fi
   if [[ $TRAVIS_BRANCH == 'release' && $TRAVIS_PULL_REQUEST == 'false' ]]; then
```
Comment (gaspard) [inline .travis.yml]: Not sure about the "release" branch. I changed this weeks ago but I think we reverted to using the "master".

Comment (gaspard) [inline .travis.yml]: Yes @edgesoft confirms, we should do the automatic release from "master". So this can be merged with the previous `if`.

Comment (Guria) [inline .travis.yml]: hmm. I don't like having any logic in travis.yml scripts now when we have repo-cooker.
I'd like to see here only repo-cooker script invocation. Th rest of the logic shoul go into actions.

Comment (gaspard) [inline .travis.yml]: Yes, seems like a good idea. We then need a new action in repo-cooker to get name of current branch.

File: package.json [hunk 2]
```diff
@@ -45,6 +44,7 @@
     "raf": "3.3.0",
     "react": "15.4.2",
     "react-dom": "15.4.2",
+    "repo-cooker": "^1.0.0",
```
Comment (gaspard) [inline package.json]: This looks gooooooood !! :-) haha 👍

**Comment by gaspard** [comment]
```yaml
# no more before_install
after_success:
- |
  if [[ $TRAVIS_BRANCH == 'master' || $TRAVIS_PULL_REQUEST == 'true' ]]; then
    npm run coverage;
    npm run coverage:upload;
  fi
  if [[ $TRAVIS_BRANCH == 'master' || $TRAVIS_BRANCH == 'next' ]] && [[ $TRAVIS_PULL_REQUEST == 'false' ]]; then
    echo "//registry.npmjs.org/:_authToken=\${NPM_TOKEN}" > .npmrc
    npm run release -- --publish
    # Maybe deploy could be part of the repo-cooker script...
    # git remote add auth https://cerebraljs:${GH_TOKEN}@github.com/cerebral/cerebral;
    # git config --global user.email "cerebraljs@gmail.com";
    # git config --global user.name "Cerebral JS";
    # npm run deploy && echo 'Website updated.';
  fi
```

File: scripts/release/index.js [hunk 3]
```diff
@@ -16,14 +13,33 @@ cooker.run('publish', [
   cook.relatedPackagesByPackage,
   cook.getCurrentVersionByPackage,
   cook.evaluateNewVersionByPackage,
+  cook.byBranch,
```
Comment (Guria) [inline scripts/release/index.js]: should getLatestReleaseHash depend on branch? how we deal with it for different branches?

File: scripts/release/index.js [hunk 4]
```diff
@@ -16,14 +13,33 @@ cooker.run('publish', [
...
+  {
+    next: cook.remap(
+      'newVersionByPackage',
+      (version, props) => `${version}-${props.hash.slice(0, 6)}`
```
Comment (Guria) [inline scripts/release/index.js]: let use datetime here, because hash caused problems with semver

File: scripts/release/index.js (hunk 3 above)
Comment (gaspard) [inline scripts/release/index.js]: We base all the @next versions on "latest" so this works correctly: we take latest "master release", compute semver and then apply semver on top of latest "master release version". Good for me.

File: scripts/release/index.js (hunk 4 above)
Comment (gaspard) [inline scripts/release/index.js]: Even a short hash ? So I propose to use

```js
version => `${version}-${new Date().getTime()}`
```

Comment (gaspard) [inline scripts/release/index.js]: I liked the hash idea... What is wrong with hash ?

Comment (Guria) [inline scripts/release/index.js]: it causes problem with ~ and ^ version resolution.

File: scripts/release/index.js (hunk 3 above)
Comment (Guria) [inline scripts/release/index.js]: next can be released several times before getting into master (aka latest)

Comment (Guria) [inline scripts/release/index.js]: ahh, got it. it's only to evaluate correct version. ok then

File: scripts/release/index.js (hunk 4 above)
Comment (christianalfoni) [inline scripts/release/index.js]: Updated to datime

**Pull request merged**